islandora_checksum_checker.module

```diff
@@ -278,8 +281,13 @@ function islandora_checksum_checker_validate_checksum($pid) {
               '!message' => $e->getMessage(),
             ),
             WATCHDOG_WARNING);
+          $exceptions[] = $e;
+          module_invoke_all('islandora_checksum_checker_outcome', $pid, $ds->id, array(), '', $exceptions);
           return FALSE;
         }
+
+        module_invoke_all('islandora_checksum_checker_outcome', $pid, $ds->id, $ds_info, $outcome, $exceptions);
```

Comment:
Hi, sorry for being such a "picky" one. The fix is good but was wondering if the code would look cleaner if instead of writing the same 3x, we could just accumulate the response (what we return FALSE| TRUE) and just have on return (where the  ```return TRUE```) is? Also. All these returns (probably diverging, but seems like the original logic could have an issue) could be ignored at all if tuque fails. Mostly sure that in a ```try {}```execution stops if something fails, and nothing can ensure you that i will reach to the invoke at all. means this need to go also (or instead if my first pickiness is well receibed) into the a catch?

Reply:
@DiegoPino sure, accumulating the response and returning once at the end sounds a bit cleaner. However, I'm not completely clear on what you're saying about the try/catch - should the connection to tuque be also wrapped in a try/catch?

Reply:
@DiegoPino looking into implementing your suggestion, I'm finding that we need to accumulate not only a TRUE/FALSE on the validation attempt, but also the `$outcome` and `$ds_info` variables for each of the three conditions. I'm not convinced doing all that to be able to call `module_invoke_all()` once is any cleaner than just invoking `module_invoke_all()` locally for each of the three conditions, as the code stands now.

But on your point about tuque failing, are you suggesting that we wrap `$tuque = islandora_get_tuque_connection($user);` in an a check? That function catches an exception and [returns NULL on failure](https://github.com/Islandora/islandora/blob/7.x/islandora.module#L1172) so perhaps we can check for NULL and  throw our own exception to pass to `module_invoke_all()`? That would be consistent with how the hook definition, allowing modules to react to a tuque failure.